testing/http_authentication.rst

```diff
@@ -4,26 +4,24 @@
 How to Simulate HTTP Authentication in a Functional Test
 ========================================================
 
-If your application needs HTTP authentication, pass the username and password
-as server variables to ``createClient()``::
+Authenticating requests in functional tests can slow down the entire test suite.
+This could become an issue especially when the tests reproduce the same steps
+that users follow to authenticate, such as submitting a login form or using
+OAuth authentication services.
 
-    $client = static::createClient(array(), array(
-        'PHP_AUTH_USER' => 'username',
-        'PHP_AUTH_PW'   => 'pa$$word',
-    ));
+This article explains the two most popular techniques to avoid these issues and
+create fast tests when using authentication.
 
-You can also override it on a per request basis::
+Using a Faster Authentication Mechanism Only for Tests
+------------------------------------------------------
 
-    $client->request('DELETE', '/post/12', array(), array(), array(
-        'PHP_AUTH_USER' => 'username',
-        'PHP_AUTH_PW'   => 'pa$$word',
-    ));
+When your application is using a ``form_login`` authentication, you can make
+your tests faster by allowing them to use HTTP authentication. This way your
+tests authenticate with the simple and fast HTTP Basic method whilst your real
```

Comment:
I treat them as synonyms. I think this is correct 🤓  but let's wait for more opinions.